### DISCUSSION TITLE
The 'Live' indicator on the admin bar is unnecessary and distracting

### DISCUSSION THREAD

**Discussion by lumpysimon** [woocommerce/woocommerce#51401] [Coming Soon Mode]
Hi

I hope this is the right place to ask about this, please let me know if it should be posted somewhere else.

Two of my clients have contacted me to ask about the new 'Live' badge on the admin toolbar. The reasons being:

 - It's suddenly appeared without explanation and wasn't there before,
 - It is showing on their password-protected staging sites, which worried one of them that their staging site was somehow 'live' to the whole world,
 - It's on every page so they didn't know it was related just to their store, not to their whole site.

To me it feels extremely distracting as it stands out more than anything else in the admin bar, so your eye is drawn to it on every single page across the entire front-end and back-end, even though most of those pages are not related to the store. 

I also feel it's unnecessary: high visibility badges like this are great for things that change a lot (e.g. for showing a number indicating how many new messages or notifications there are), but this is something that is likely to change just once in the lifetime of a store.

The options I have thought of are:

1. Show a badge only when the store is in Coming Soon mode, then show nothing when the store is live.
2. Show it only on store pages on the front-end, and only on WooCommerce admin pages.
3. Provide a true/false filter in the `site_visibility_badge` function so displaying the badge can be overridden.

Would you consider any/all of those options?

**Comment by Graphnic** [comment]
It doesn't seem to be an eCommerce feature per-se, and is quite intrusive both in terms of location and visual appearance.  It's also potentially misleading when a site is, in fact, in staging as @lumpysimon mentions.

It's not for me to comment on whether the feature should be part of WooCommerce, but in my opinion, it should default off and have a toggle control in the settings.

**Comment by ChadReitsma** [comment]
I agree, it's extremely distracting. There are also _**several**_ other "Coming soon" and "Maintenance" plugins available. @lumpysimon's suggestions are excellent.

**Comment by adrianduffell** [comment]
Thanks for the feedback, you found the right place!

I think we made a mistake releasing the badge without a choice to remove it. We would like to address this by adding a feature toggle in advanced settings to remove it.

<img width="1792" alt="site vis disabled" src="https://github.com/user-attachments/assets/ddc709f7-101d-4530-b369-610670f32f09">

Now that it has been released to all users, we are seeing significant use of the badge in live mode where users are trying out the coming soon mode. We would like to give users the choice now whether to keep the badge or remove it with this setting.

Please let me know what you think.

We would also like to address the confusion when sites are in a staging environment. We are interested whether it could be solved by checking [wp_get_environment_type](https://developer.wordpress.org/reference/functions/wp_get_environment_type/) and changing the label when `staging` is returned. Is this environment type commonly used in your staging environments?

![image](https://github.com/user-attachments/assets/7a15dac4-a740-4f06-bf82-84fa947e1f0c)

**Reply by Drivingralle** [reply]
I always use the environment type. The websites don't need to display it's state

**Comment by Graphnic** [comment]
A check box is perfect!  Personally, I often forget to use wp_get_environment_type when creating a manual staging site, but I can see that's a sensible thing to use for this purpose.

**Comment by scottbuscemi** [comment]
+1'ing the concerns in this thread. I was alarmed to see my subscriptions "staging" site is "Live"!

I would advocate for having this opt-in rather than opt-out. If a checkbox for opting out is needed, the "Site Visibility" page is quite barren so it could go there rather than buried in the Advanced > Features page.

Lastly, it's the "WordPress Way" to have filters available for new features like this. Also +1'ing the suggestion of having a filter to hide this.
